bundles/org.openhab.binding.rachio/src/main/java/org/openhab/binding/rachio/internal/discovery/RachioDiscoveryService.java

```diff
@@ -0,0 +1,190 @@
+    protected synchronized void startScan() {
+        Future<?> scanTask = this.scanTask;
+        if (scanTask == null || scanTask.isDone()) {
+            logger.debug("Starting Rachio discovery scan");
+            scanTask = scheduler.submit(this::discover);
+        }
+    }
+
+    protected synchronized void discover() {
+        try {
+            RachioBridgeHandler handler = cloudHandler;
+            if (handler == null) {
+                logger.debug("RachioDiscovery: Rachio Cloud access not set!");
+                return;
+            }
+
+            HashMap<String, RachioDevice> deviceList = null;
+            ThingUID bridgeUID;
+            deviceList = handler.getDevices();
+            bridgeUID = handler.getThing().getUID();
+
+            if (deviceList == null) {
+                logger.debug("Discovery: Rachio Cloud access not initialized yet!");
+                return;
+            }
+            logger.debug("Found {} devices.", deviceList.size());
+            for (HashMap.Entry<String, RachioDevice> de : deviceList.entrySet()) {
+                RachioDevice dev = de.getValue();
+                logger.debug("Check Rachio device with ID '{}'", dev.id);
+
+                // register thing if it not already exists
+                ThingUID devThingUID = new ThingUID(THING_TYPE_DEVICE, bridgeUID, dev.getThingID());
+                dev.setUID(bridgeUID, devThingUID);
+                logger.info(" Rachio device discovered: '{}' (id {}), S/N={}, MAC={}", dev.name, dev.id,
+                        dev.serialNumber, dev.macAddress);
+                logger.debug("  latitude={}, longitude={}", dev.latitude, dev.longitude);
+                logger.info("   device status={}, paused/sleep={}, on={}", dev.status, dev.getSleepMode(),
+                        dev.getEnabled());
+                Map<String, Object> properties = new HashMap<>(dev.fillProperties());
+                DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(devThingUID).withProperties(properties)
+                        .withBridge(bridgeUID).withLabel(dev.getThingName()).build();
+                thingDiscovered(discoveryResult);
+
+                HashMap<String, RachioZone> zoneList = dev.getZones();
+                logger.debug("Found {} zones for this device.", zoneList.size());
+                for (HashMap.Entry<String, RachioZone> ze : zoneList.entrySet()) {
+                    RachioZone zone = ze.getValue();
+                    logger.debug("Checking zone with ID '{}'", zone.id);
+
+                    // register thing if it not already exists
+                    ThingUID zoneThingUID = new ThingUID(THING_TYPE_ZONE, bridgeUID, zone.getThingID());
+                    zone.setUID(devThingUID, zoneThingUID);
+                    logger.info("Zone#{} '{}' (id={}) added, enabled={}", zone.zoneNumber, zone.name, zone.id,
+                            zone.getEnabled());
+
+                    if (zone.getEnabled() == OnOffType.ON) {
```

Comment:
Shouldn't all zones be added, even if they are currently be disabled?  The zones can be ignored in the inbox if needed.